internal/controller/maintenancemode_controller.go

```diff
@@ -64,8 +64,10 @@ func (r *MaintenanceModeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			&v1alpha1.StorageClient{},
 			&handler.EnqueueRequestForObject{},
 			builder.WithPredicates(
-				generationChangePredicate,
-				maintenanceModeChangedPredicate,
+				predicate.Or(
+					generationChangePredicate,
+					maintenanceModeChangedPredicate,
+				),
```

Comment:
not sure about this, we usually want to skip generation changes for objects to which we are only interested in spec changes and since you already have a predicate for MMChanged the other predicate isn't required.

Reply:
ack makes sense, I will update it